third_party/ros/roslaunch/BUILD.bazel

```diff
@@ -1,21 +1,36 @@
 load("@rules_python//python:defs.bzl", "py_binary", "py_library")
 load("@rules_ros_pip_deps//:requirements.bzl", "requirement")
 load("//build_tools:pylint.bzl", "pylint")
-load("//third_party:expand_template.bzl", "expand_template")
 
-expand_template(
+genrule(
     name = "deps",
-    out = "deps.py",
-    data = ["roscore.xml"],
-    substitutions = {"{roscore_xml_path}": "$(rootpath roscore.xml)"},
-    template = "deps.py.tpl",
+    outs = ["deps.py"],
+    cmd = """cat <<EOF > $@
+ROSMASTER_PATH = '$(rootpath @ros_comm//:rosmaster)'
+ROSCORE_XML_PATH = '$(rootpath roscore.xml)'
+EOF""",
+    srcs = ["roscore.xml"],
+    tools = ["@ros_comm//:rosmaster"],
+)
+
+genrule(
+    name = "roscore_xml",
+    outs = ["roscore.xml"],
+    cmd = """cat <<EOF > $@
+<launch>
+  <group ns="/">
+    <node type="'$(rootpath @ros_comm//:rosout)'" name="rosout" respawn="true"/>
+  </group>
+</launch>
+EOF""",
+    tools = ["@ros_comm//:rosout"],
```

Comment:
Similarly here: rosout should be IMO in srcs, if it's in tools then the target is going to be built in exec config.